test/apps/iodemo/io_demo.cc

```diff
@@ -2010,6 +2062,12 @@ class DemoClient : public P2pDemoCommon {
         _server_info[server_index].active_index = _active_servers.size() - 1;
         active_servers_make_used(_server_info[server_index].active_index);
         assert(_num_active_servers_to_use <= _active_servers.size());
+
+        // Place the new connection in a random active index
+        size_t active_index =
+                IoDemoRandom::rand<size_t>(_next_active_index,
```

Comment:
why not rand from 0?

Reply:
when adding the new server, we want tp handle it in the round.
so, need to insert it in random place with the index from `[_next_active_index, _num_active_servers_to_use)`

Reply:
why place it in this round? IMO better not limit the place we put it in to have more randomness

Reply:
> why place it in this round? IMO better not limit the place we put it in to have more randomness

but then, it will be not visited in this round - so, it will report `min: 0`. if it can be problematic, if connection will be failed/restored - the `min: 0` will appear for all logs which report performance.

Reply:
but even if placed in this round, the op will not complete immediately?

Reply:
> but even if placed in this round, the op will not complete immediately?

yes, we will wait for it prior reporting performance data

Reply:
/* Insert new connection into random index and it can send in either this round or next round */

Reply:
done